p2p/net/connmgr/connmgr.go

```diff
@@ -239,6 +240,13 @@ func (cm *BasicConnMgr) IsProtected(id peer.ID, tag string) (protected bool) {
 	return protected
 }
 
+func (cm *BasicConnMgr) CheckLimit(systemLimit connmgr.GetterConnLimit) error {
+	if cm.cfg.highWater > systemLimit.GetConnLimit() {
+		return fmt.Errorf("conn manager high water limit: %d, exceeds the system limit of: %d", cm.cfg.highWater, systemLimit.GetConnLimit())
```

Comment:
NIT: I'd prefer: 
>conn manager high watermark limit: %d, exceeds the system connection limit of: %d

Though this is fine too since from connmgr it can be inferred that the issue is relating to connection limits.